contracts/src/instances/ezeth/ezETHBase.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IERC20 } from "../../interfaces/IERC20.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IRestakeManager } from "../../interfaces/IRestakeManager.sol";
+import { HyperdriveBase } from "../../internal/HyperdriveBase.sol";
+import { FixedPointMath, ONE } from "../../libraries/FixedPointMath.sol";
+
+/// @author DELV
+/// @title ezETH Base Contract
+/// @notice The base contract for the ezETH Hyperdrive implementation.
+/// @dev
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract EzETHBase is HyperdriveBase {
+    using FixedPointMath for uint256;
+
+    /// @dev The Renzo entrypoint contract.
+    IRestakeManager internal immutable _restakeManager;
+
+    /// @dev The ezETH token contract.
+    IERC20 internal immutable _ezETH;
+
+    /// @notice Instantiates the ezETH Hyperdrive base contract.
+    /// @param __restakeManager The Renzo Restakemanager contract.
+    constructor(IRestakeManager __restakeManager) {
+        _restakeManager = __restakeManager;
+        (, bytes memory data) = address(__restakeManager).call(
+            abi.encodeWithSignature("ezETH()")
+        );
+        _ezETH = IERC20(abi.decode(data, (address)));
+    }
+
+    /// Yield Source ///
+
+    /// @dev Accepts a transfer from the user in base token.
+    /// @param _baseAmount The base amount to deposit.
+    /// @return sharesMinted The shares that were minted in the deposit.
+    /// @return refund The amount of ETH to refund. This should be zero for
+    ///         yield sources that don't accept ETH.
+    function _depositWithBase(
+        uint256 _baseAmount,
+        bytes calldata // unused
+    ) internal override returns (uint256 sharesMinted, uint256 refund) {
+        // Ensure that sufficient ether was provided.
+        if (msg.value < _baseAmount) {
+            revert IHyperdrive.TransferFailed();
+        }
+
+        // If the user sent more ether than the amount specified, refund the
+        // excess ether.
+        unchecked {
+            refund = msg.value - _baseAmount;
+        }
+
+        uint256 balanceBefore = _ezETH.balanceOf(address(this));
+        // Submit the provided ether to Renzo to be deposited.
+        // a referrer id can be put in here
```

Comment:
Move the comment to the top of the code block (right before `uint256 balanceBefore). We should remove the sentence about the referral ID since it isn't being included. We should also explain how we compute shares minted.